src/Resources/ResourceManager/Implementation/Deployments/GetAzureDeploymentCmdlet.cs

```diff
@@ -12,49 +12,144 @@
+        [Parameter(Position = 0, ParameterSetName = GetAzureDeploymentCmdlet.ResourceGroupParameterSetWithDeploymentName, Mandatory = true,
+            HelpMessage = "The scope type of the deployment.")]
+        [Parameter(Position = 0, ParameterSetName = GetAzureDeploymentCmdlet.SubscriptionParameterSetWithDeploymentName, Mandatory = true,
+            HelpMessage = "The scope type of the deployment.")]
+        [Parameter(Position = 0, ParameterSetName = GetAzureDeploymentCmdlet.ManagementGroupParameterSetWithDeploymentName, Mandatory = true,
+            HelpMessage = "The scope type of the deployment.")]
+        [Parameter(Position = 0, ParameterSetName = GetAzureDeploymentCmdlet.TenantParameterSetWithDeploymentName, Mandatory = true,
+            HelpMessage = "The scope type of the deployment.")]
+        [ValidateNotNullOrEmpty]
+        public DeploymentScopeType? ScopeType { get; set; }
+
+        [Parameter(Position = 1, ParameterSetName = GetAzureDeploymentCmdlet.ManagementGroupParameterSetWithDeploymentName, Mandatory = true,
+            HelpMessage = "The management group id.")]
+        [ValidateNotNullOrEmpty]
+        public string ManagementGroupId { get; set; }
+
+        [Parameter(Position = 1, ParameterSetName = GetAzureDeploymentCmdlet.ResourceGroupParameterSetWithDeploymentName, Mandatory = true,
+            HelpMessage = "The resource group name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
 
         [Alias("DeploymentName")]
-        [Parameter(Position = 0, ParameterSetName = GetAzureDeploymentCmdlet.DeploymentNameParameterSet, Mandatory = false,
+        [Parameter(Position = 1, ParameterSetName = GetAzureDeploymentCmdlet.TenantParameterSetWithDeploymentName, Mandatory = false,
+            HelpMessage = "The name of deployment.")]
+        [Parameter(Position = 2, ParameterSetName = GetAzureDeploymentCmdlet.ManagementGroupParameterSetWithDeploymentName, Mandatory = false,
+            HelpMessage = "The name of deployment.")]
+        [Parameter(Position = 1, ParameterSetName = GetAzureDeploymentCmdlet.SubscriptionParameterSetWithDeploymentName, Mandatory = false,
+            HelpMessage = "The name of deployment.")]
+        [Parameter(Position = 2, ParameterSetName = GetAzureDeploymentCmdlet.ResourceGroupParameterSetWithDeploymentName, Mandatory = false,
             HelpMessage = "The name of deployment.")]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
         [Alias("DeploymentId", "ResourceId")]
-        [Parameter(ParameterSetName = GetAzureDeploymentCmdlet.DeploymentIdParameterSet, Mandatory = false,
+        [Parameter(ParameterSetName = GetAzureDeploymentCmdlet.DeploymentIdParameterSet, Mandatory = true,
             HelpMessage = "The fully qualified resource Id of the deployment. example: /subscriptions/{subId}/providers/Microsoft.Resources/deployments/{deploymentName}")]
```

Comment:
In that sense it's actually breaking. In the past you can provide neither name nor Id, which gets all the deployments at subscription scope. Now it would not be possible, you will be asked to provide name/scopeType if you run the cmdlet without any parameter.